src/lib.rs

```diff
@@ -68,15 +77,15 @@ impl Into<Elem> for Index {
 
 impl From<Elem> for Index {
   fn from(e: Elem) -> Self {
-    Self::of32(e>>16, e>>8 & 0xff, e&0xff)
+    Self::of32(e>>16, e>>8 & 0xff, e&0xff).unwrap()
   }
 }
 
 trait StoreCloneable {
   fn duplicate(&self) -> Self;
 }
 
-#[derive(Default, Debug)]
+#[derive(Default, Debug, PartialEq)]
```

Comment:
Don't love this for the same reason I don't love deriving `clone`, if we have `==` at all it should use structural equality. Does `assert_eq!(x.0, y.0)` work?